### ISSUE TITLE
Fails on Linux 3.13 (Ubuntu 14.04 kernel in semaphore) due to missing `renameat2()`

### ISSUE THREAD

**Issue by martinpitt** [systemd/casync#36]
Some tests fail in semaphore as the kernel is rather old there. While Ubuntu 14.04 offers newer kernels, there is no way to install/reboot into them on semaphore, so we have to make-do.

strace can't decipher the syscall unfortunately, but  the errno is "Function not implemented".
```
$ strace -fvvs1024 ./test-casync
[...]
syscall_316(0xffffff9c, 0x7fff0eee8990, 0xffffff9c, 0x7fff0eee8920, 0x1, 0x10eee8b80) = -1 (errno 38)
unlinkat(AT_FDCWD, "/var/tmp/teststore.90853f1e88177e2f/5473/54737747f8c1f940744a24119640f36973f79e2a4367fafea1b1d65d4c213ed3.ba0d1fbaf6ce9875.tmp", 0) = 0
unlinkat(AT_FDCWD, "/var/tmp/teststore.90853f1e88177e2f/5473", AT_REMOVEDIR) = 0
write(2, "Assertion failed: r >= 0\n", 25Assertion failed: r >= 0
```

The other tests also fail with "Function not implemented", so at first sight it's the same issue.

**Comment by martinpitt** [comment]
Ah, syscall 316 is `renameat2()`, unfortunately that's one of the crucial ones, no?

**Comment by poettering** [comment]
Should be fixed by 0be064e33d523654a1b7ff75092c05529dcc80b0

**Comment by martinpitt** [comment]
Fixed with https://github.com/systemd/casync/commit/0be064e33d523654a
